### PR TITLE
Save 100's of KB's in bundle-size by removing the global lodash dependency

### PR DESCRIPTION
**Pull request by kiwi-josh**
## Description of the change

This PR's removes the global lodash package - as this pulls in all of lodash and add 100's of KBs to the bundle size of the rudderstack SDK

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/667)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,9 +49,14 @@
     "crypto-js": "3.1.9-1",
     "get-value": "^3.0.1",
     "is": "^3.3.0",
-    "lodash": "^4.17.21",
     "lodash.clonedeep": "^4.5.0",
```
Comment (kiwi-josh) [inline package.json:52]: Specific packages were already used previously, but at some point the global `lodash` package was added, which pulls in every single lodash function 😱  a lot of which aren't needed.

Comment (saikumarrs) [reply package.json:52]: Yes, it has been addressed for the core SDK, but integrations were left out.

**Review by utsabc — APPROVED** [review]

**Review by saikumarrs — COMMENTED** [review]
Please update the limit in `.size-limit.js` to `510 kB`.

**Comment by kiwi-josh** [comment]
@saikumarrs I have updated the `.size-limit.js` to `510KB` as requested

**Comment by saikumarrs** [comment]
@kiwi-josh Thanks for incorporating the suggestions.
As a next step, please sign the CLA using this [form](https://rudderlabs.wufoo.com/forms/rudderlabs-contributor-license-agreement).

**Review by ItsSudip — APPROVED** [review]

**Comment by MoumitaM** [comment]
Hey @kiwi-josh, Have you signed the CLA?
As soon as you do that we will absorb this change.

**Comment by kiwi-josh** [comment]
@MoumitaM done

**Comment by MoumitaM** [comment]
@kiwi-josh, Please update the `.size-limit.js` to `1150` kb.

**Comment by kiwi-josh** [comment]
@MoumitaM done. I am happy for rudderstack to take over authoring this PR and format it / make it suitable for deploy

**Review by MoumitaM — APPROVED** [review]

**Review by shrouti1507 — DISMISSED** [review]

**Comment by MoumitaM** [comment]
@kiwi-josh, Can you resolve the conflicts?

**Comment by kiwi-josh** [comment]
@MoumitaM done

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**